contracts/contracts/strategies/balancer/BalancerComposablePoolStrategy.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Balancer ComposableStablePool Strategy
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { BalancerMetaPoolStrategy } from "./BalancerMetaPoolStrategy.sol";
+import { IBalancerVault } from "../../interfaces/balancer/IBalancerVault.sol";
+import { IERC20 } from "../../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+
+contract BalancerComposablePoolStrategy is BalancerMetaPoolStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
+    /* @notice position of BPT token in the Balancer's pool
+     *
+     * @dev this could be a storage variable that gets set by reading Balancer's pool
+     * tokens and comparing it to platformAddress. Seems more convenient, but we rather
+     * go for gas savings of an immutable variable.
+     */
+    uint256 public immutable bptTokenPoolPosition;
+
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        BaseBalancerConfig memory _balancerConfig,
+        address _auraRewardPoolAddress,
+        uint256 _bptTokenPoolPosition
+    )
+        BalancerMetaPoolStrategy(
+            _stratConfig,
+            _balancerConfig,
+            _auraRewardPoolAddress
+        )
+    {
+        bptTokenPoolPosition = _bptTokenPoolPosition;
+    }
+
+    /*
+     * @dev Value that represents exit encoding where for max BPT given
+     * request exactly specifies the amount of underlying assets
+     * to be returned.
+     */
+    function _btpInExactTokensOutIndex()
+        internal
+        pure
+        override
+        returns (uint8)
+    {
+        return
+            uint8(
+                IBalancerVault
+                    .ComposablePoolExitKind
+                    .BPT_IN_FOR_EXACT_TOKENS_OUT
+            );
+    }
+
+    /*
+     * @dev Value that represents exit encoding where BPT tokens are supplied for
+     * proportional exit is required when calling a withdrawAll.
+     */
+    function _exactBptInTokensOutIndex()
+        internal
+        pure
+        override
+        returns (uint8)
+    {
+        return
+            uint8(
+                IBalancerVault
+                    .ComposablePoolExitKind
+                    .EXACT_BPT_IN_FOR_ALL_TOKENS_OUT
+            );
+    }
+
+    function _assetConfigVerification(address[] calldata _assets)
+        internal
+        view
+        override
+    {
+        require(
+            // aside from BPT token all assets must be supported
+            poolAssets.length - 1 == _assets.length,
+            "Pool assets length mismatch"
+        );
+
+        require(
+            // BPT position should be correctly configured
+            poolAssets[bptTokenPoolPosition] == platformAddress,
+            "BPT token position incorrect"
+        );
+
+        for (uint256 i = 0; i < _assets.length; ++i) {
```

Comment:
That is a great point thanks. I have just read up on this how; Solidity (after 0.8.0) doesn't need to perform overflow checks in situations where no overflow is guaranteed. And looping through arrays definitely is one of those situations.

The unchecked does provide some gas savings with a little minor impact on code readability. In this case I would rather not make the change, since first loop is done only on initialisation. And the other 2 are done on join/exit pools. We are running those operations relatively rarely - and are paying for the gas ourselves.

If these optimisations would be part of `checkBalance` function ( that gets called in Vault's allocate/rebase that are called in most of mints & redeems) then I'd gladly trade the readability impact for the gas savings.